src/parser/shaman/shared/maelstromchart/Maelstrom.js

```diff
@@ -32,83 +22,91 @@ const Maelstrom = props => {
     );
   }
 
-  const maxResource = props.tracker.maxResource || props.max;
   const { start } = props;
 
-  const resource = [];
-  const waste = [];
-
+  const rawData = [];
 
   props.tracker.resourceUpdates.forEach((item) => {
     const secIntoFight = Math.floor((item.timestamp - start) / 1000);
-    resource.push({x: secIntoFight, y:item.current});
-    waste.push({x: secIntoFight, y:item.waste});
+    rawData.push({kind: 'Maelstrom', x: secIntoFight, y:item.current});
+    rawData.push({kind: 'Wasted', x: secIntoFight, y:item.waste});
   });
 
+  const data = {
+    data: rawData,
+  };
+
```

Comment:
Something akin to this is seemingly everywhere - can it be made more generic so that only select changes need to be passed to a chart module?

Reply:
I would say it is probably not a great idea. I did that for the footerchart specifically but we will likely need to undo it in the future. Vega supports multiple datasets as input, a feature that I used repeatedly in converting charts. For example: the Stagger chart has 4 datasets as input: hp, stagger, purifies, and deaths.

While the `BaseChart` *could* infer that array input = single dataset, I think it is more trouble than its worth.

Reply:
Fair!